mdx/guides/swap-tokens-with-0x-api.mdx

```diff
@@ -0,0 +1,278 @@
+
+(async () => {
+    // buying 1 DAI with USDC
+    const params = {
+        buyToken: 'DAI',
+        sellToken: 'USDC',
+        buyAmount: '1000000000000000000',
+    }
+    
+    const response = await fetch(
+        `https://api.0x.org/swap/v0/quote?${qs.stringify(params)}`
+        );
+    
+    await window.web3.eth.sendTransaction(await response.json());
+})();
+```
+
+## Specifying a taker address for your swaps
+
+You can optionally provide a `takerAddress` to get a specific gas amount to send with the swap transaction. 
+
+The API will also perform a `eth_call` to catch any reverts that would occur if the `takerAddress` attempts to swap the tokens. An HTTP response with status `400` will be returned if the eth_call results in a revert (ie. the swap fails), along with reasons for the revert.
+
+```javascript 
+import * as qs from 'qs';
+
+(async () => {
+    // selling 1 ETH for DAI
+    const params = {
+        buyToken: 'DAI',
+        sellToken: 'ETH',
+        sellAmount: '1000000000000000000',
+        takerAddress: '0xAb5801a7D398351b8bE11C439e05C5B3259aeC9B', // vitalik's address
+    }
+
+    const response = await fetch(
+        `https://api.0x.org/swap/v0/quote?${qs.stringify(params)}`
+        );
+    const responseJson = await response.json();
+    if (response.ok) {
+        await window.web3.eth.sendTransaction(responseJson);
+    } else {
+        // Contains the error
+        console.log(responseJson);
+    }
+})();
+```
+
+## Diagnosing issues with reverts
+
+Here is a handy checklist to go through to diagnose a revert issue:
+
+* Does the `takerAddress` have enough balance? Check by comparing the `sellAmount` provided in the response of the endpoint against the `takerAddress`‘s `sellToken` balance.
+* If `sellToken` is a ERC20 token, (not ETH), the `takerAddress` will need to approve [0x ERC20Proxy](https://etherscan.io/address/0x95e6f48254609a6ee006f7d493c8e5fb97094cef).
+
+## Seeing tokens available for swapping
+
+The `/swap/v0/tokens` endpoint is useful to see a registry of all tokens and their respective token symbols available to swap through `/swap/v0/quote`. We maintain and update a registry of all the tokens (off-chain) and their token symbols available through `/swap/v0/quote`. Feel free to make edits or additions in the open-source repo [here](https://github.com/0xProject/0x-api).
+
+```javascript
+(async () => {
+    const response = await fetch('https://api.0x.org/swap/v0/tokens');
+        
+    const tokens = (await response.json()).records;
+    
+    console.log(`${JSON.stringify(tokens, null, 2)}`);
+})();
+```
+
+Logs:
+
+```json
+[
+    {
+        "symbol": "DAI",
+        "address": "0x6b175474e89094c44da98b954eedeac495271d0f"
+     },
+     {
+        "symbol": "REP",
+        "address": "0x1985365e9f78359a9B6AD760e32412f4a445E862"
+     },
+     {
+        "symbol": "WETH",
+        "address": "0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2"
+     },
+     ...
+]
+```
```

Comment:
thinking outloud: Might be worth introducing pagination here (this won't grab all the token pairs as we have more than 20) in a second variation

Reply:
this endpoint isn't actually paginated (yet)